docs/luos-technology/services/routing-table.md

```diff
@@ -37,12 +37,20 @@ When each service in the network has an attributed ID, the detection algorithm p
 
 Sometimes, multiple services in the network can have the same alias, which is not allowed to prevent service confusion. In this case, the detection algorithm will add a number after each alias instance on the routing table.
 
+At the end of the detection process, each service receives a message with the command of `END_DETECTION`, that can be used in your code in order to understand the exact moment that the detection has finished, and you can then reinitialize the behavior of your services (for example reinitialize the auto-updates, etc,). Also, you can check the detection status of your node, by using the dedicated Luos API:
+
+```c
+Luos_IsNodeDetected();
+```
+ 
+that returns `true` or `false` depending on whether this node is detected or not.
+
 :::caution Warnings
 
 1. Be careful that a service can change ID during a detection depending on the service running this detection.
 2. Do not consider your service's ID fixed.
 3. Be aware that every service removes its auto-update configuration during the detection to prevent any ID movement.
-
+4. Make sure that by the creation of your services you specify a callback pointer for each of them, so that the messages arriving to them, concerning the detection do not stay stored in your node's memory.
```

Comment:
I don't understand this: the detection fails if services doesn't have a callback declared ?

Reply:
No it will work but if it is a service with no callback and with no messages pulling the end_detection message will stay stored in the buffer and at a moment or an other the node will crush

Reply:
Ok. I think it's a problem, don't you ? We should be able to run a system containing services without a callback.

Reply:
we are able. but the user needs to pull the messages in order to empty the memory. In any way the only case that this may cause a probleme is the case that a user creates an application that only sends messages. And this is a risk that we are willing to take in order to use the end_detection message. Anyways, in the creation of services we say to associate a callback with the service, the only user that will not do that is an already advanced user, who is able to understand where the messages will go and that he needs to pull the messages in order to empty any message that maybe came to the network.

Reply:
It's a design choice but I think this could leads to problems for users. I often use services which doesn't use callbacks, they only send messages to others services as you mentionned. As a developper, I wouldn't understand why I should create an empty function to pass to luos_CreateService if I don't use it. 
I think it leads to more difficulties to understand the technology and the code.

Reply:
it is not the only case that your service will fail, and end detection is not the only message that can arrive to the service without expecting it. For example, if you use a gate, and you have created your custom made application of your own custom type, your gate will be demanding updates to your service because by default, it demands updates to all the sensors and to all the types that it doesnt recognize. So the need of a callback, implies a means of good development practice and not a patch to save your node in case of end_detection. I think this notification should have been in the doc earlier than the end detection

Reply:
Ok I understand.